pkg/apis/options/providers.go

```diff
@@ -0,0 +1,177 @@
+package options
+
+import "github.com/oauth2-proxy/oauth2-proxy/v7/providers"
+
+// Providers is a collection of definitions for providers.
+type Providers []Provider
+
+// Provider holds all configuration for a single provider
+type Provider struct {
+	// ClientID is the OAuth Client ID that is defined in the provider
+	// This value is required for all providers.
+	ClientID string `json:"clientID,omitempty"`
+	// ClientSecret is the OAuth Client Secret that is defined in the provider
+	// This value is required for all providers.
+	ClientSecret string `json:"clientSecret,omitempty"`
+	// ClientSecretFile is the name of the file
+	// containing the OAuth Client Secret, it will be used if ClientSecret is not set.
+	ClientSecretFile string `json:"clientSecretFile,omitempty"`
```

Comment:
I wonder if we should make this a `SecretSource`, could be a follow up to convert to minimise the diff, but I think that would help long term

Reply:
yep, I would prefer this to be a follow up, since it is not logically related to the multiple providers changes and also like you said to minimise the diff